### PR TITLE
Automated test report generation

### PR DESCRIPTION
**Pull request by gkellogg**
This may need some permissions added to the repo, but should cause test manifests and reports to be automatically updated in the PR if anything in /tests or /reports is modified.

### CODE REVIEW

**Comment by gkellogg** [comment]
@jeswr, took some time off and on, but I found a pattern that simplified the task. Thanks for all your suggestions.

File: .github/workflows/generate.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  build:
+    name: Generate Reports using Ruby
+    runs-on: ubuntu-latest
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
```
Comment (jeswr) [inline .github/workflows/generate.yml:18]: I suspect you will still need to supply the `GH_TOKEN` here and in the commit action (see https://github.com/eyereasoner/eye-js/blob/a0008130be3389ddf473943dbdcc7c8aaf57943a/.github/workflows/update-eye.yml#L9-L11)

Comment (gkellogg) [reply .github/workflows/generate.yml:18]: The `permissions` key above seemed to do it, as I didn't need it to get it working in my fork, and it did add the automated commit to this PR.

Comment (jeswr) [reply .github/workflows/generate.yml:18]: LGTM then :)

**Comment by peacekeeper** [comment]
Great to see some work on automated tests, thanks!

**Review by iherman — APPROVED** [review]

**Review by dlongley — APPROVED** [review]

**Pull request merged**